File: 26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        count = Counter(nums)
        demo = []
        for i in range(len(nums)):
            if count[nums[i]] > 0:
                demo.append(nums[i])
                count[nums[i]] = 0
        nums[:] = demo[:]
        return len(demo)

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int beg = 0, end = 0;
        
        while (end < n) {
            if (nums[beg] != nums[end])
                nums[++beg] = nums[end];
            end++;
        }
        return beg + 1;
    }
};